scripts/release/create-release.hs

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env cabal 
+{- project:
+  with-compiler: ghc-9.4
+-}
+{- cabal:
+  build-depends:
+    base,
+    commonmark,
+    containers,
+    filepath,
+    foldl,
+    text,
+    turtle ^>=1.6.0,
+-}
+{-# OPTIONS_GHC -Wall -Wextra #-}
+{-# LANGUAGE GHC2021 #-}
+{-# LANGUAGE BlockArguments, LambdaCase, OverloadedStrings, ScopedTypeVariables #-}
+
+import Commonmark
+import Control.Monad
+import Data.Map (Map)
+import Data.Monoid (First(..))
+import Data.Semigroup (Max(..))
+import Data.Version
+import System.FilePath
+import Turtle hiding (fp, d, o)
+import qualified Data.Map as Map
+import qualified Data.Text as Text
+import qualified Data.Text.IO as Text
+import qualified Control.Foldl as Foldl
+
+main :: IO ()
+main = sh do
+
+  (isDryRun, skipGit) <- options "Create releases for ouroboros-consensus packages" $
+    (,) <$> switch "dry-run" 'd' "Make no changes"
+        <*> switch "skip-git" 's' "Skip creating a new release branch or commits for these changes"
+
+  -- Determine how breaking the sum of the changes in each package is
+  packageChangeSeverities <- reduce collectSeverities do
+
+    (packageName, dependencies) <- select packages
+
+    maxMaybe <- reduce Foldl.maximum do
+      fragment <- findChangelogFragments packageName
+      findChangeSeverity fragment
+
+    pure (packageName, maxMaybe, dependencies)
+
+  packageVersions <- reduce Foldl.map do
+    (packageName, maybeSeverity) <- select $ Map.toList packageChangeSeverities
+
+    case maybeSeverity of
+      Nothing -> do
+        liftIO do
+          putStrLn $ "No changes need to be made for package " <> packageName <> "!"
+        mzero
+      Just severity -> do
+        currentPackageVersion <- findCurrentPackageVersion packageName
+        let nextPackageVersion = calculateNextPackageVersion severity currentPackageVersion
+
+        pure (packageName, (currentPackageVersion, nextPackageVersion))
+
+  void $ liftIO $ flip Map.traverseWithKey packageVersions $ \package (current, next) -> do
+    putStrLn $ package <> ": " <> showVersion current <> " -> " <> showVersion next
+
+  unless isDryRun do
+    unless skipGit do
+      createGitBranch packageVersions
+
+  (packageName, (current, next)) <- select $ Map.toList packageVersions
+
+  if isDryRun
+    then
+      liftIO do
+        putStrLn $ "This is a dry run, so no changes will be made for " <> packageName
+    else do
+      updateCabalFile packageName current next packageVersions
+      runScrivCollect packageName next
+      unless skipGit do
+        createGitCommit packageName next
+
+-- | Pairs of packages with a list of their dependencies. We use the
+--   dependencies to calculate which version changes should require a version
+--   bump in a package's dependencies as well.
+--
+--   BEWARE! This list should always be ordered such that any given package's
+--   dependencies are located BEFORE that package in the list!
+packages :: [(FilePath, [FilePath])]
+packages =
```

Comment:
Can't this list be auto-generated? `stack dot` I think is able to generate a graph of deps so there must be some way to do it